### PR TITLE
feat: implement `debug_getRawHeader`

### LINKED ISSUE #328: Hive Testing: `debug_getRawHeader`
- [ ] get-block-n
- [ ] get-genesis
- [ ] get-invalid-number

### PR DESCRIPTION
**Pull request by JulianVentura**
**Motivation**

Currently we are missing some RPC debug calls. This PR aims to implement one of them.

**Description**

This PR implement RPC call `debug_getRawHeader`.

**Changes**

* Added struct `GetRawHeaderRequest` with `RpcHandler` trait implementation
* Modified `RpcHandler::parse` trait function to return `RpcErr` instead of `None` on error case:
    * This had to be done in order to return an specific error structure `RpcErr::BadHexFormat` to fix failing hive tests.
    * Refactor has been made on all structs implementing this trait
* Added new RPC namespace `RpcNamespace::Debug` for debug RPCs.

Resolves #328
Closes #328

### CODE REVIEW

File: crates/rpc/utils.rs [hunk 1]
```diff
@@ -34,6 +35,11 @@ impl From<RpcErr> for RpcErrorMetadata {
                 data: None,
                 message: "Unsupported fork".to_string(),
             },
+            RpcErr::BadHexFormat => RpcErrorMetadata {
+                code: -32602,
+                data: None,
+                message: "invalid argument 0: hex string without 0x prefix".to_string(),
```
Comment (avilagaston9) [inline crates/rpc/utils.rs:41]: I guess the error won't always be in argument 0. Maybe we can add a field to `BadHexFormat` to store the index of the argument?

File: crates/rpc/eth/block.rs [hunk 2]
```diff
@@ -193,6 +198,43 @@ impl RpcHandler for GetBlockReceiptsRequest {
...
+        };
+
+        let block: BlockIdentifier = serde_json::from_value(params[0].clone())?;
+        let block_str: String = serde_json::from_value(params[0].clone())?;
+
+        if let BlockIdentifier::Number(_) = block {
+            if !block_str.starts_with("0x") {
+                return Err(RpcErr::BadHexFormat);
+            }
+        }
```
Comment (avilagaston9) [inline crates/rpc/eth/block.rs:215]: Not for this PR, but I think we should add this check to all the others block requests too

**Review by avilagaston9 — APPROVED** [review]
LGTM! With just a nit comment

File: crates/rpc/utils.rs (hunk 1 above)
Comment (JulianVentura) [reply crates/rpc/utils.rs:41]: Yes, that makes sense. I'll make the change

File: crates/rpc/eth/block.rs [hunk 3]
```diff
@@ -193,6 +198,43 @@ impl RpcHandler for GetBlockReceiptsRequest {
...
+        };
+
+        let block: BlockIdentifier = serde_json::from_value(params[0].clone())?;
+        let block_str: String = serde_json::from_value(params[0].clone())?;
+
+        if let BlockIdentifier::Number(_) = block {
+            if !block_str.starts_with("0x") {
+                return Err(RpcErr::BadHexFormat(0));
+            }
+        }
```
Comment (fmoletta) [inline crates/rpc/eth/block.rs:215]: This looks like a check that should be done for all cases where `BlockIdentifier` is used. Maybe we could add a method for `BlockIndentifier` that takes care of this check while parsing (such as `parse(serde_value: Value) -> Result<BlockIdentifier, RpcErr>`) that we can use for all cases without repeating code

**Pull request merged**